Sources/Screens/SignOutConfirmation/WalletSignOutPageViewModel.swift

```diff
@@ -0,0 +1,76 @@
+import GDSAnalytics
+import GDSCommon
+import Logging
+import UIKit
+
+struct WalletSignOutPageViewModel: GDSInstructionsViewModel, BaseViewModel {
+    let title: GDSLocalisedString = "app_signOutConfirmationTitle"
+    let body: String = GDSLocalisedString(stringLiteral: "app_signOutConfirmationBody1").value
+    var childView = UIView()
+    let buttonViewModel: any ButtonViewModel
+    let secondaryButtonViewModel: (any ButtonViewModel)? = nil
+    let analyticsService: AnalyticsService
+    
+    let rightBarButtonTitle: GDSLocalisedString? = "app_cancelButton"
+    let backButtonIsHidden: Bool = true
+    
+    init(analyticsService: AnalyticsService,
+         buttonAction: @escaping () -> Void) {
+        self.analyticsService = analyticsService
+        self.buttonViewModel = AnalyticsButtonViewModel(titleKey: "app_signOutAndDeleteAppDataButton",
+                                                        backgroundColor: .gdsRed,
+                                                        analyticsService: analyticsService) {
+            buttonAction()
+        }
+        self.childView = configureStackView()
+    }
+    
+    func didAppear() {
+        let screen = ScreenView(id: ProfileAnalyticsScreenID.signOutScreenWithWallet.rawValue,
+                                screen: ProfileAnalyticsScreen.signOutScreenWithWallet,
+                                titleKey: title.stringKey)
+        analyticsService.trackScreen(screen)
+    }
+    
+    func didDismiss() {
+        let event = ButtonEvent(textKey: "back")
+        analyticsService.logEvent(event)
```

Comment:
@dylanjames-dd is this quite right?
I think this will get called anytime that that screen disappears regardless of whether the user has tapped "back" or not.

Reply:
Should be [only triggered from the right bar button item](https://github.com/govuk-one-login/mobile-ios-common/blob/main/Sources/GDSCommon/Patterns/BaseViewController.swift#L28-L55) unless the action from there is called by the OS?

Reply:
Ah wonderful 🙂 ignore me completely then!

Reply:
Does pose the question of whether a swipe down to dismiss this modal screen should be calling this or some other kind of analytics event 🤔